src/pyspiffe/svid/jwt_svid_validator.py

```diff
@@ -0,0 +1,118 @@
+import datetime
+from calendar import timegm
+
+from pyspiffe.svid.exceptions import (
+    TokenExpiredError,
+    InvalidClaimError,
+    InvalidAlgorithmError,
+    InvalidTypeError,
+)
+
+AUDIENCE_NOT_MATCH_ERROR = 'audience does not match expected value'
+INVALID_INPUT_ERROR = 'Invalid input: {}.'
+
+
+class JwtSvidValidator(object):
+    """Performs validations on a given token checking compliance to SPIFFE specification.
+    See <a href="https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md">https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md</a>.
+    """
+
+    _required_claims = ['aud', 'exp', 'sub']
+    _supported_algorithms = [
+        'RS256',
+        'RS384',
+        'RS512',
+        'ES256',
+        'ES384',
+        'ES512',
+        'PS256',
+        'PS384',
+        'PS512',
+    ]
+    _supported_types = ['JWT', 'JOSE']
+
+    def __init__(self) -> None:
+        pass
+
+    """
+    Validates token header by verifing if header specifies supported algortihms and token type. Type is optional but in case it is present, it must be set to the supported values.
+
+    Args:
+        header ({}): token header.
+
+    Returns:
+        None.
+
+    Raises:
+        ValueError: in case header is not specified.
+        InvalidAlgorithmError: in case specified 'alg' is not supported as specified by the SPIFFE standard.
+        InvalidTypeError: in case 'typ' is present in header but is not set to 'JWT' or 'JOSE'.
+    """
+
+    def validate_header(self, header: {}) -> None:
+        if not header:
+            raise ValueError(INVALID_INPUT_ERROR.format('header cannot be empty'))
+
+        if header['alg'] not in self._supported_algorithms:
+            raise InvalidAlgorithmError(header['alg'])
+        try:
+            if header['typ'] and header['typ'] not in self._supported_types:
+                raise InvalidTypeError(header['typ'])
+        except KeyError:
+            pass
```

Comment:
Due to the check `if header['alg']` I assume that the dict `header` might not contain all the keys. In that case I think we need to used `dict.get(key, default_value)` here. It will return the value for the given `key` at the dictionary or a `default_value` which by default is `None`. So here `header.get('alg')` implicitly handles `KeyError` exception.

I think this needs to be rewritten as:
```
   alg = header.get('alg')
   if not alg:
       raise INVALID_INPUT_ERROR.format('header alg cannot be empty.')
   if alg not in self._supported_algorithms:
       raise InvalidAlgorithmError(alg)

   typ = header.get('typ')
   if not typ:
       raise INVALID_INPUT_ERROR.format('header type cannot be empty.')
       
    if typ not in self._supported_types:
        raise InvalidTypeError(typ)
```
Another option could be to just raise the error and not make any difference between an invalid and an empty alg/typ.

Reply:
Thanks Andres for the tip! I changed the code to use .get method. It is much better!

Reply:
Ah! I also changed the other occurrences of the usage of Dict too. Thanks!